.github/scripts/strategies/syntax/rubocop.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# =============================================================================
+# strategies/syntax/rubocop.sh - Rubocop linter for union-merged files
+# =============================================================================
+# Runs Rubocop style checks on Ruby files changed by union merge.
+# Returns 0 (warning only) to avoid blocking workflow on style issues.
+# =============================================================================
+
+# shellcheck source=.github/scripts/lib/core.sh
+source "$(dirname "${BASH_SOURCE[0]}")/../../lib/core.sh"
+
+validate_rubocop() {
+  # Check if Rubocop is available
+  if ! command -v rubocop >/dev/null 2>&1; then
+    log_warn "Rubocop not installed, skipping linter check"
+    return 0
+  fi
+
+  log_info "Running Rubocop on changed Ruby files..."
+
+  # Get Ruby files that were modified in HEAD commit
+  local files
+  files="$(git diff --name-only HEAD~1 HEAD 2>/dev/null | grep '\.rb$' || true)"
+
+  if [[ -z "$files" ]]; then
+    log_info "No Ruby files changed"
+    return 0
+  fi
+
+  log_debug "Checking files: $files"
+
+  # Run Rubocop (respects .rubocop.yml)
+  # Output goes to logs, warnings shown in GitHub UI
+  if ! rubocop $files 2>&1; then
```

Comment:
if change to using bundled version of Rubocop, may have to use:
```shell
  if ! bundle exec rubocop $files 2>&1; then
```

Reply:
resolved in commit